### ISSUE TITLE
Possibility of eMSP having their own tariffs

### ISSUE THREAD

**Issue by ardias** [ocpi/ocpi#474]
From reading the spec and some of the discussions here, it seems that Tariffs and therefore session cost (to the drivers) is meant to always be applied in the CPO. 
Is it contemplated in the protocol that an eMSP might have different tariffs than the CPO and presents those tariffs to drivers (via their app or webpage) and therefore charges drivers using that tariff and ignores the CPO tariff? 
It raises some issues, but the use case seems valid.  I (as an eMSP) use some CPO for handling the infrastructure side of things (physical chargers, etc) but then I present the drivers with a tariff I define myself. It could be based on the Tariff the CPO has defined, or be completely different, with different prices , different restrictions, etc.. I could completely ignore what the CPO charges me (which seems to be what the CPO tariffs and CDRs are meant for, to charge the eMSP) and make sessions free or make them cost double, add service fees, etc..

Let's imagine a scenario: I as driver am subscribed to some eMSP service that uses one or many CPOs support. Each CPO might have a tariff to charge some amount for energy consumed. But since I have a subscription with an eMSP, I could be charged nothing for individual charging sessions or I could be charged something completely different , either more or less.  Or could be the other way around, the CPO is "publicly subsidized" so there is no cost, but the eMSP does want to charge based on time and/or energy.  And I (eMSP) still want to show "realtime" updated on energy consumption and session cost to my drivers.

While trying to implement something like this, it seems the information being passed from the CPO to the eMSP (session data, for example) is not enough to be able to apply a Tariff properly (tariff similar to CPO tariffs), since we do not have access to the meter data. The charging periods only contain the volume and the start time of the period, so trying to figure out what price component to apply seems impossible, especially if the charger disconnects for some time, and send meter data with times in the past. The CPO has access to this info an knows what tariff component to apply, but this info gets lost when the session is pushed to the eMSP (if it ever gets pushed).

Has anyone thought about or implemented something like this, meaning, charging drivers with tariffs defined in the eMSP?

**Comment by ardias** [comment]
@andacata Yes, that's my understanding. The problem is the (at least as I see it) lack of enough information sent to the eMSP to be able to apply a similar tariff as those in the CPO

**Comment by andacata** [comment]
I know some eMSPs that are applying a % over the CPO price, or offering a flat rate to the drivers.

**Comment by richardjfleming** [comment]
If this is the case, what is the view of adding some extra information in the session updates going from CPO to eMSP in order to be able to introduce more complex B2C tariffs in the eMSP. In my case I need the latest charging period update timestamp  from the CPO to be able to do my own eMSP charging period generation (based on eMSP tariffs).
